src/annotator/test/config-test.js

```diff
@@ -1,86 +1,220 @@
 'use strict';
 
-var config = require('../config');
-var unroll = require('../../shared/test/util').unroll;
-
-describe('annotator configuration', function () {
-  var fakeScriptConfig;
-
-  function fakeQuerySelector(selector) {
-    if (selector === 'link[type="application/annotator+html"]') {
-      return {href: 'app.html'};
-    } else if (selector === 'script.js-hypothesis-config' &&
-               fakeScriptConfig) {
-      return {textContent: fakeScriptConfig};
-    } else {
-      return null;
-    }
-  }
-
-  var fakeWindowBase = {
+var proxyquire = require('proxyquire');
+
+var fakeSettings = sinon.stub();
+var fakeExtractAnnotationQuery = {};
+
+var config = proxyquire('../config', {
+  '../shared/settings': fakeSettings,
+  './util/extract-annotation-query': fakeExtractAnnotationQuery,
+});
+var sandbox = sinon.sandbox.create();
+
+function fakeWindow() {
+  return {
     document: {
-      querySelector: fakeQuerySelector,
-      querySelectorAll: function (selector) {
-        var match = fakeQuerySelector(selector);
-        return match ? [match] : [];
-      },
+      querySelector: sinon.stub().returns({href: 'LINK_HREF'}),
     },
-    location: {hash: ''},
+    location: {href: 'LOCATION_HREF'},
   };
+}
 
-  beforeEach(function () {
-    fakeScriptConfig = '';
+describe('src.annotator.config', function() {
+  beforeEach('stub console.warn()', function() {
+    sandbox.stub(console, 'warn');
   });
 
-  it('reads the app src from the link tag', function () {
-    var linkEl = document.createElement('link');
-    linkEl.type = 'application/annotator+html';
-    linkEl.href = 'https://test.hypothes.is/app.html';
-    document.head.appendChild(linkEl);
-    assert.deepEqual(config(window), {
-      app: linkEl.href,
+  beforeEach('reset fakeSettings', function() {
+    fakeSettings.reset();
+    fakeSettings.returns({});
+  });
+
+  beforeEach('reset fakeExtractAnnotationQuery', function() {
+    fakeExtractAnnotationQuery.extractAnnotationQuery = sinon.stub();
+  });
+
+  afterEach('reset the sandbox', function() {
+    sandbox.restore();
+  });
+
+  context("when there's an application/annotator+html <link>", function() {
+    var link;
+
+    beforeEach('add an application/annotator+html <link>', function() {
+      link = document.createElement('link');
+      link.type = 'application/annotator+html';
+      link.href = 'http://example.com/link';
+      document.head.appendChild(link);
+    });
+
+    afterEach('tidy up the link', function() {
+      document.head.removeChild(link);
```

Comment:
This fixes an issue with the old test that if the test failed the link was left in the document